### PR TITLE
feat(core): custom icon placement on core button

### PR DESCRIPTION
**Pull request by platon-rov**
## Related Issue(s)

Relates to https://github.com/SAP/fundamental-ngx/issues/9011

## Description

Feature to have custom icon or any content on button **before** the text.

## Screenshots

CleanShot 2022-12-14 at 18 51 20@2x

### CODE REVIEW

File: libs/core/src/lib/button/button.component.html [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
+<ng-container *ngIf="glyphPosition === 'after'">
+    <ng-container [ngTemplateOutlet]="iconTemplate"></ng-container>
 </ng-container>
 
 <fd-icon glyph="slim-arrow-down" *ngIf="fdMenu"></fd-icon>
+
+<ng-template #iconTemplate>
+    <fd-icon *ngIf="glyph" [glyph]="glyph"></fd-icon>
+
+    <ng-content></ng-content>
```
Comment (N1XUS) [inline libs/core/src/lib/button/button.component.html:18]: Maybe keep original ng-content, and use this one with [select]?
There might be cases when button is used like so:
```
<button fd-button>
   Some inner text
   <fd-icon...></fd-icon>
</buton>```

**Review by N1XUS — CHANGES_REQUESTED** [review]

File: libs/core/src/lib/button/button.component.html (hunk 1 above)
Comment (platon-rov) [reply libs/core/src/lib/button/button.component.html:18]: Adding `[select]` is fine, though it adds one more detail to think of. 

Your example won't be affected with current changes, but still no any new details added.

Comment (platon-rov) [reply libs/core/src/lib/button/button.component.html:18]: `glyphPosition` is `before` by default, so we show projected content even if nothing (like in your example) specified, neither glyph, label or glyphPosition.

**Comment by InnaAtanasova** [comment]
Correct me if I am wrong, but our buttons have the option to add an icon. The problem the user is facing is not having access to the button in Action Sheet, which prevents them from adding an icon.
Screenshot 2022-12-15 at 11 36 07 AM

**Review by mikerodonnell89 — APPROVED** [review]

**Pull request merged**